src/model/Model.js

```diff
@@ -270,13 +270,36 @@ export default class Model {
 		return new KeyModel( escapeKey( this.key ) );
 	}
 
-	getKeypathModel () {
-		return this.keypathModel || ( this.keypathModel = new KeypathModel( this ) );
+	getKeypathModel ( ractive ) {
+		let keypath = this.getKeypath(), model = this.keypathModel || ( this.keypathModel = new KeypathModel( this ) );
+
+		if ( ractive && ractive.component ) {
+			let mapped = this.getKeypath( ractive );
+			if ( mapped !== keypath ) {
```

Comment:
Got it. Took a look at the tests though, we need to check mapping though two levels of components.